servers/physics/physics_server_sw.cpp

```diff
@@ -192,11 +192,14 @@ real_t PhysicsServerSW::space_get_param(RID p_space, SpaceParameter p_param) con
 	return space->get_param(p_param);
 }
 
-PhysicsDirectSpaceState *PhysicsServerSW::space_get_direct_state(RID p_space) {
+PhysicsDirectSpaceState *PhysicsServerSW::space_get_direct_state(RID p_space, bool p_flag_error) {
 
 	SpaceSW *space = space_owner.getornull(p_space);
-	ERR_FAIL_COND_V(!space, NULL);
-	ERR_FAIL_COND_V_MSG(!doing_sync || space->is_locked(), NULL, "Space state is inaccessible right now, wait for iteration or physics process notification.");
+
+	if (p_flag_error) {
```

Comment:
I don't understand why this `p_flag_error` argument is added.

Not running `ERR_FAIL_COND_V(!space, NULL)` means crashing with a null pointer dereference when the condition is actually valid.

Reply:
Ah yes you are right. I probably just need to mask out the doing_sync and is_locked.

Sorry I'm having problems with git at the moment, will update it as soon as I can work out how git works lol. :smile: